### PR TITLE
Fix data loss when writing sorted Hive tables

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

**Comment by martint** [comment]
Can you explain what’s the bug in the commit message? It’s not entirely obvious from the diff.

File: presto-hive/src/main/java/com/facebook/presto/hive/SortingFileWriter.java [hunk 1]
```diff
@@ -193,7 +192,7 @@ private void combineFiles()
             int count = min(maxOpenTempFiles, tempFiles.size() - (maxOpenTempFiles - 1));
 
             List<TempFile> smallestFiles = IntStream.range(0, count)
-                    .mapToObj(i -> tempFiles.pollFirst())
+                    .mapToObj(i -> tempFiles.poll())
```
Comment (findepi) [inline presto-hive/src/main/java/com/facebook/presto/hive/SortingFileWriter.java]: `tempFiles.remove()` would be better, as empty queue is an exceptional condition

Comment (electrum) [reply presto-hive/src/main/java/com/facebook/presto/hive/SortingFileWriter.java]: Good point, I forgot about that method. Though we will fail here with NPE due to immutable list.

Comment (findepi) [reply presto-hive/src/main/java/com/facebook/presto/hive/SortingFileWriter.java]: I know. Just that NPE is slightly more confusing that NoElement... Surely by looking into the code one will know why NPE.

**Pull request merged**